tests/config_test.go

```diff
@@ -346,3 +348,70 @@ config:
 	assert.Equal(t, "[\"third\"]", thirdValue["value"])
 	assert.Equal(t, []interface{}{"third"}, thirdValue["objectValue"])
 }
+
+func TestConfigCommandsUsingEnvironments(t *testing.T) {
+	if getTestOrg() != pulumiTestOrg {
+		t.Skip("Skipping test because the required environment is in the moolumi org.")
+	}
+	t.Parallel()
+
+	e := ptesting.NewEnvironment(t)
+	defer func() {
+		deleteIfNotFailed(e)
+	}()
+
+	integration.CreateBasicPulumiRepo(e)
+	e.RunCommand("pulumi", "org", "set-default", getTestOrg())
+	e.RunCommand("pulumi", "stack", "init", "test")
+
+	// check config is empty
+	stdout, _ := e.RunCommand("pulumi", "config")
+	assert.Equal(t, "KEY  VALUE", strings.Trim(stdout, "\r\n"))
+
+	// set an esc environment
+	e.RunCommand("pulumi", "config", "env", "add", "secrets-test-env-DO-NOT-DELETE", "--yes")
+
+	// just `pulumi config`
+	stdout, _ = e.RunCommand("pulumi", "config")
+	assert.Equal(t, `KEY          VALUE
+test_secret  [secret]`, strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config --show-secrets`
+	stdout, _ = e.RunCommand("pulumi", "config", "--show-secrets")
+	assert.Equal(t, `KEY          VALUE
+test_secret  this_is_my_secret`, strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config --open`
+	stdout, _ = e.RunCommand("pulumi", "config", "--open")
+	assert.Equal(t, `KEY          VALUE
+test_secret  [secret]`, strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config --show-secrets --open`
+	stdout, _ = e.RunCommand("pulumi", "config", "--show-secrets", "--open")
+	assert.Equal(t, `KEY          VALUE
+test_secret  this_is_my_secret`, strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config --show-secrets --open=false`
+	stdout, _ = e.RunCommand("pulumi", "config", "--show-secrets", "--open=false")
+	assert.Equal(t, `KEY          VALUE
+test_secret  [unknown]`, strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config get`
+	stdout, _ = e.RunCommand("pulumi", "config", "get", "test_secret")
+	assert.Equal(t, "this_is_my_secret", strings.Trim(stdout, "\r\n"))
+
+	// `pulumi config get --open=false`
+	stdout, _ = e.RunCommand("pulumi", "config", "get", "test_secret", "--open=false")
+	assert.Equal(t, "[unknown]", strings.Trim(stdout, "\r\n"))
+
+	// delete the stack
+	e.RunCommand("pulumi", "stack", "rm", "-s", "test", "--yes")
+}
+
+func getTestOrg() string {
+	testOrg := pulumiTestOrg
+	if _, set := os.LookupEnv("PULUMI_TEST_ORG"); set {
+		testOrg = os.Getenv("PULUMI_TEST_ORG")
```

Comment:
```suggestion
	if org, set := os.LookupEnv("PULUMI_TEST_ORG"); set {
		return org
```
No need to look it up again.